BlogTemplate/Pages/Account/ConfirmEmail.cshtml.cs

```diff
@@ -28,13 +28,13 @@ public ConfirmEmailModel(UserManager<ApplicationUser> userManager)
             var user = await _userManager.FindByIdAsync(userId);
             if (user == null)
             {
-                throw new ApplicationException($"Unable to load user with ID '{userId}'.");
+                throw new FieldAccessException($"Unable to load user with ID '{userId}'.");

```

Comment:
I don't believe this is the right exception type. This exception is for rare cases in reflection where invalid code was emitted into a DLL and the .NET runtime has to throw an exception.

You could instead:
1. throw an `InvalidOperationException()` - this is a catch-all type of exception that just means "something bad happened so I'm gonna throw a tantrum" 😢 😿 😭 
2. Better yet, at least in controllers and pages, instead of throwing exceptions, which isn't meaningful (the end-user just sees an HTTP 500 error with no details), I recommend returning an appropriate Action Result that reflects the specific problem. For example, in this case here, you could return a `NotFound` result (i.e. `return NotFound();`) to indicate that something wasn't found (a user). In each other controller error case you could find an appropriate action result that reflects the HTTP status code that is appropriate. If there are non-controller (or non-page) cases, we can look to see what an appropriate error or exception is.

From what I see in this PR, unless I missed something, I think option (2) is the way to go. It's a best practice for web apps to return appropriate HTTP status codes.